force-app/main/default/classes/RD2_ValidationService.cls

```diff
@@ -180,32 +180,44 @@ public with sharing class RD2_ValidationService {
 
         Boolean isNew = oldRds == null || oldRds.isEmpty();
 
-        if (isNew) {
-            errorRds = validateOnInsert();
-
-        } else {
-            errorRds = validateOnChange();
-        }
+        errorRds = validateOnInsertOrUpdate(isNew);
 
         return errorRds;
     }
 
     /**
     * @description Validate RD data on insert
+    * @param isNew True if the Recurring Donation is New, False if not
     * @return void
     */
-    public List<ErrorRecord> validateOnInsert() {
+    public List<ErrorRecord> validateOnInsertOrUpdate(Boolean isNew) {
         List<ErrorRecord> errorRecords = new List<ErrorRecord>();
 
-        for (npe03__Recurring_Donation__c rd : rds) {
+        for (Integer i = 0; i < rds.size(); i++) {
+            npe03__Recurring_Donation__c rd = rds[i];
+            npe03__Recurring_Donation__c oldRd = isNew ? null : oldRds[i];
             ErrorRecord errorCollection = new ErrorRecord(rd);
 
-            validateActiveRecurringDonation(rd, errorCollection);
-            validateInstallmentFrequency(rd, errorCollection);
-            validateDonor(rd, accountById.get(rd.npe03__Organization__c), errorCollection);
-            validateStatus(rd, errorCollection);
+            RD2_RecurringDonation recurringDonation = new RD2_RecurringDonation(rd);
+            if (!recurringDonation.isClosed()) {
+                validateAmount(rd, errorCollection);
```

Comment:
Not a show-stopper, but it might be nice to abstract these checks into another method.  Maybe validateOpenRDFields or something to indicate all unclosed RDs are included.